java/templates/s2i/assemble

```diff
@@ -74,6 +74,10 @@ function copy_artifacts() {
 }
 
 function setup_maven() {
+  if [ -f "${S2I_SOURCE_DIR}/configuration/settings.xml" ]; then
+    cp ${S2I_SOURCE_DIR}/configuration/settings.xml $HOME/.m2/settings.xml
+    echo "Using custom maven settings from ${S2I_SOURCE_DIR}/configuration/settings.xml"
+  fi
```

Comment:
now that you moved this up, could you please add a check for the `sed` command below which replaced the proxy placeholder, that indeed a placeholder is included in the (possibly custom) settings.xml ? If not, a warning should be printed out like in 

"HTTP proxy environment variables have been provided but no placeholder '<!-- ### configured http proxy ### -->' found in ${S2I_SOURCE_DIR}/configuration/settings.xml. Ignoring the HTTP proxy setup for now"

The check could be done by a simple grep before the sed.

Otherwise, the PR is good to merge.